coc/raid.py

```diff
@@ -0,0 +1,515 @@
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+import typing
+from typing import List, TYPE_CHECKING
+
+from . import BasePlayer
+from .miscmodels import Badge, Timestamp, try_enum
+from .utils import cached_property, correct_tag
+
+if TYPE_CHECKING:
+    # pylint: disable=cyclic-import
+    from .war_members import ClanWarMember  # noqa
+
+
+class RaidMember(BasePlayer):
+    """Represents a Raid Member that the API returns.
+
+    Attributes
+    ----------
+    tag: :class:`str`
+        The player's tag
+    name: :class:`str`
+        The player's name
+    attack_count: :class:`int`
+        The number of attacks from this player
+    attack_limit: :class:`int`
+        The limit of attacks from this player
+    bonus_attack_limit: :class:`int`
+        The limit of bonus attacks from this player
+    capital_resources_looted:  :class:`int`
+        The amount of resources looted by this player
+    raid_log_entry: :class:`RaidLogEntry`
+        The raid log entry this member is in
+    """
+
+    __slots__ = ("tag",
+                 "name",
+                 "attack_count",
+                 "attack_limit",
+                 "bonus_attack_limit",
+                 "capital_resources_looted",
+                 "raid_log_entry",
+                 "_attacks",
+                 "_client")
+
+    def __init__(self, *, data, client, raid_log_entry):
+        super().__init__(data=data, client=client)
+        self._client = client
+        self.raid_log_entry = raid_log_entry
+        self._attacks = []
+        self._from_data(data)
+
+    def __repr__(self):
+        attrs = [
+            ("tag", self.tag),
+            ("raid_log_entry", repr(self.raid_log_entry)),
+            ("attack_count", self.attack_count)
+        ]
+        return "<%s %s>" % (self.__class__.__name__, " ".join("%s=%r" % t for t in attrs),)
+
+    def __eq__(self, other):
+        return (isinstance(other, RaidMember)
+                and self.tag == other.tag
+                and self.raid_log_entry == other.raid_log_entry
+                and self.attacks == other.attacks)
+
+    def _from_data(self, data):
+        data_get = data.get
+
+        self.name: str = data_get("name")
+        self.tag: str = data_get("tag")
+        self.attack_count: int = data_get("attacks")
+        self.attack_limit: int = data_get("attackLimit")
+        self.bonus_attack_limit: int = data_get("bonusAttackLimit")
+        self.capital_resources_looted: int = data_get("capitalResourcesLooted")
```

Comment:
`.get` by default returns `None` if not found, so typing needs some care

Reply:
as this is the way coc.py does it everywhere else, i would stick with it